src/components/progress/progress.tsx

```diff
@@ -148,40 +95,37 @@ export const EuiProgress: FunctionComponent<ExclusiveUnion<
   ...rest
 }) => {
   const determinate = !isNil(max);
-  let colorClass = null;
-  let dataColorClass = null;
-  let optionalCustomStyles: any = null;
-  if (color) {
-    if (isNamedColor(color)) {
-      colorClass = colorToClassNameMap[color];
-      dataColorClass = dataColorToClassNameMap[color];
-    } else {
-      optionalCustomStyles = { color: color };
-      colorClass = 'euiProgress--customColor';
-    }
-  }
-  const classes = classNames(
-    'euiProgress',
-    {
-      'euiProgress--indeterminate': !determinate,
-      'euiProgress--native': determinate,
-    },
-    sizeToClassNameMap[size],
-    colorClass,
-    positionsToClassNameMap[position],
-    className
-  );
-  const dataClasses = classNames(
-    'euiProgress__data',
-    {
-      'euiProgress__data--l': size === 'l',
-    },
-    dataColorClass
-  );
-  const labelClasses = classNames(
-    'euiProgress__label',
-    labelProps && labelProps.className
-  );
+
+  const isNamedColor = COLORS.includes(color as EuiProgressColor);
+  const optionalCustomStyles = !isNamedColor ? { color } : {};
```

Comment:
> I don't think so, primarily because the custom color is getting set as a style prop and not passed through to emotion.

You can still use our color functions to affect inline styles, they're not restricted to Emotion:

```suggestion
  const optionalCustomStyles = !isNamedColor ? { makeHighContrastColor(color)(euiTheme) } : {};
```

Reply:
Ahhh!! Thank you!!

Reply:
Now I'm wondering if I should have used `makeHighContrastColor()` for EuiTextColor and custom colors. Maybe not since consumers probably expect the color they pass to be the one used?

Reply:
Yeah that's a bit different since they're specifically specifiying the text `color` versus in this component they're telling us what color they want the progress bar to be and we're syncing it with the text color

Reply:
https://github.com/elastic/eui/pull/5986/commits/9959c4f404ea2aca0921718b8d9ce6e831ce96c9

<img width="372" alt="" src="https://user-images.githubusercontent.com/549407/175349718-465b1a9b-7a97-4a3f-bedf-dfcf91e348f6.png">

Looking great! Just for clarity as well, I assumed the bar color should not need the high contrast function, just the text, but please shout if that assumption was incorrect.